### ISSUE TITLE
[SCM-989] Tests fail if svn and/or git are not installed

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1219]
**[Niels Basjes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nielsbasjes)** opened **[SCM-989](https://issues.apache.org/jira/browse/SCM-989?redirect=false)** and commented

If svn is not installled (As is the case in Ubuntu in WSL2 in Windows 11) the build fails over these tests:

```java
[ERROR] Errors:
[ERROR]   SvnRemoteInfoCommandTest.testExist:42 » Scm Error while executing svn command.
[ERROR]   SvnRemoteInfoCommandTest.testNotExist:53 » Scm Error while executing svn command.
```

---

**Remote Links:**
- [GitHub Pull Request #149
](https://github.com/apache/maven-scm/pull/149)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

```
[INFO]
[ERROR] Failures:
[ERROR]   SvnExeListCommandTckTest>ScmTckTestCase.setUp:144->SvnListCommandTckTest.initRepo:47->ScmTestCase.execute:266 Exit value wasn't 0, was:1
[ERROR]   SvnExeListCommandTckTest>ScmTckTestCase.setUp:144->SvnListCommandTckTest.initRepo:47->ScmTestCase.execute:266 Exit value wasn't 0, was:1
[ERROR]   SvnExeListCommandTckTest>ScmTckTestCase.setUp:144->SvnListCommandTckTest.initRepo:47->ScmTestCase.execute:266 Exit value wasn't 0, was:1
[ERROR] Errors:
[ERROR]   SvnRemoteInfoCommandTest.testExist:42 � Scm cmd.exe /X /C "svn --non-interacti...
[ERROR]   SvnRemoteInfoCommandTest.testNotExist:53 � Scm cmd.exe /X /C "svn --non-intera...
[INFO]
[ERROR] Tests run: 93, Failures: 3, Errors: 2, Skipped: 29
```

and

```
[ERROR] Errors:
[ERROR]   GitInfoCommandTest.testInfoCommand:52 � ScmRepository Error creating the scm r...
[ERROR]   GitInfoCommandTest.testInfoCommandWithNegativeShortRevision:85 � ScmRepository
[ERROR]   GitInfoCommandTest.testInfoCommandWithShortRevision:68 � ScmRepository Error c...
[ERROR]   GitInfoCommandTest.testInfoCommandWithZeroShortRevision:102 � ScmRepository Er...
[INFO]
[ERROR] Tests run: 95, Failures: 0, Errors: 4, Skipped: 21
```

**Comment by jira-importer** [comment]
**[Niels Basjes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nielsbasjes)** commented

I just ran a check and indeed the same happens if git is missing.

I found the same list you just posted.

Perhaps the check if the commandline tool has been installed check should be moved 1 class up so it can be used in all tests?

Or are these tests really TCK tests and should be changed in that direction?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This, I need to check. Frankly, I don't know the reason between TCK and UT.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [262e8afc6e79b63a5c5e0e57de2cbf9434c7d7ee](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=262e8afc6e79b63a5c5e0e57de2cbf9434c7d7ee).
